src/serial_impl.rs

```diff
@@ -0,0 +1,54 @@
+//! Implementation of [`Serial`](https://docs.rs/embedded-hal/0.2.1/embedded_hal/serial/index.html)
+
+
+use nb;
+use ::hal::serial::{Read, Write};
+use ::serial;
+
+/// Newtype around [`serial::SystemPort`] that implements the `embedded-hal` traits
+pub struct Serial(pub serial::SystemPort);
+
+impl Read<u8> for Serial {
+    type Error = serial::Error;
+
+    fn read(&mut self) -> nb::Result<u8, Self::Error> {
+        use std::io::Read;
+        let mut buffer = [0; 1];
+        let bytes_read = self.0.read(&mut buffer)
+            .map_err(|err| nb::Error::Other(Self::Error::from(err)))?;
+        if bytes_read == 1 {
+            Ok(buffer[0])
+        } else {
+            Err(nb::Error::WouldBlock)
+        }
+    }
+}
+
+impl Write<u8> for Serial {
+    type Error = serial::Error;
+
+    fn write(&mut self, word: u8) -> nb::Result<(), Self::Error> {
+        use std::io::Write;
+        self.0.write(&[word])
+            .map_err(|err| nb::Error::Other(Self::Error::from(err)))?;
+        Ok(())
+    }
+
+    fn flush(&mut self) -> nb::Result<(), Self::Error> {
+        use std::io::Write;
+        self.0.flush()
+            .map_err(|err| nb::Error::Other(Self::Error::from(err)))
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_empty() {
+        let mut port: Box<Read<u8, Error=serial::Error>> = Box::new(Serial(serial::open("/dev/tty1").unwrap()));
```

Comment:
Agreed that this is not a guarantee.  For CI it might be possible to do an interesting test using a pty but even with that you might want to gate as many environments may not have permissions to setup the pty (which can be a bit of work on its own).

Reply:
Hmm that's indeed kind of a bad test. I tried to to give it a try with `openpty` from libc, which seems to work somehow: https://github.com/rust-embedded/linux-embedded-hal/pull/9/commits/176fc2876840f1bdd1af95e3f3684c512e3ae72b
Would a test like that be feasible? If yes how would one go to implement it good enough and how should I gate it?

Reply:
hmmmm, tbh i don't know anything about openpty. 
if it works i think i'd be okay with it cleaned it up and gated so it only runs on linux?

Reply:
thought tbqh i'd be just as happy with the original test `[ignore]`'d so it's opt in

Reply:
I rewrote the tests using `openpty` and they work quite well, so I consider this resolved. Or should I still gate the test?

Reply:
looks good to me, thanks ^_^